### PR TITLE
refactor: support manipulating the same data in different transaction…

### PR DESCRIPTION
**Pull request by dk-lockdown**
… branches

ref: https://github.com/cectc/dbpack/issues/199

### Ⅲ. Why don't you add test cases (unit test/integration test)? 

there is a test case:
first:
```
insertSql := fmt.Sprintf(`INSERT /*+ XID('%s') */ INTO dept_manager ( id, emp_no, dept_no, from_date, to_date ) VALUES (?, ?, ?, ?, ?)`, xid)
result, err := tx.Exec(insertSql, 1, 100001, 1001, "2022-01-01", "2024-01-01")
```
then:
```
updateDeptManagerSql := fmt.Sprintf(`UPDATE /*+ XID('%s') */ dept_manager SET to_date = ? WHERE id = ?`, xid)
result, err = tx.Exec(updateDeptManagerSql, "2026-01-01", 1)
```
they are in a global transaction, and operate the same data. check the ci result to verify this pr.

### CODE REVIEW

File: pkg/dt/storage/etcd/etcd.go [hunk 1]
```diff
@@ -129,14 +130,18 @@ func (s *store) AddBranchSession(ctx context.Context, branchSession *api.BranchS
...
+		}
 		for _, rowKey := range rowKeys {
 			comparisons = append(comparisons, notFound(rowKey))
 		}
 
 		for _, rowKey := range rowKeys {
-			lockKey := fmt.Sprintf("lk/%s/%s", branchSession.XID, rowKey)
-			ops = append(ops, clientv3.OpPut(lockKey, rowKey))
-			ops = append(ops, clientv3.OpPut(rowKey, lockKey))
+			rowKeyValue := fmt.Sprintf("lk/%s/%s", branchSession.XID, rowKey)
```
Comment (bohehe) [inline pkg/dt/storage/etcd/etcd.go]: This format `lk/%s/%s` should be a constant to make it more readable.

File: pkg/dt/storage/etcd/etcd.go [hunk 2]
```diff
@@ -495,6 +523,43 @@ func notFound(key string) clientv3.Cmp {
 	return clientv3.Compare(clientv3.ModRevision(key), "=", 0)
 }
 
+func (s *store) filterRowKey(ctx context.Context, rowKeys []string, xid string) ([]string, error) {
```
Comment (bohehe) [inline pkg/dt/storage/etcd/etcd.go]:
```suggestion
func (s *store) filterRowKeys(ctx context.Context, rowKeys []string, xid string) ([]string, error) {
```

**Review by bohehe — APPROVED** [review]
LGTM

File: pkg/dt/storage/etcd/etcd.go (hunk 1 above)
Comment (dk-lockdown) [reply pkg/dt/storage/etcd/etcd.go]: If this string template is changed to constant, and the replaced value is separated from the template, will it be difficult for people to understand?

Comment (bohehe) [reply pkg/dt/storage/etcd/etcd.go]: I mean, how about using `LockKeyFormat` constant in case other files will use this format in the future?

Comment (dk-lockdown) [reply pkg/dt/storage/etcd/etcd.go]: yes, you are right

**Review by fatelei — APPROVED** [review]

**Review by hzliangbin — APPROVED** [review]

**Review by hzliangbin — APPROVED** [review]

**Pull request merged**